### PR TITLE
PHP 8: Code Review `Database`

### PR DESCRIPTION
**Pull request by iszmais**
Hi, @chfsx 

I completed the review of the `Services/Database` component.

Results:
- [ ] The code style is `PSR-2 + X` compliant
- [x] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`
- [x] There is a `Unit Test Suite` with at least one unit test
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [ ] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or explict PHP types are used (type hints, return types, typed properties)

I also fixed some trivial inspection profile issues.

### Further actions needed:
- All unsolved Problems are marked with `//PHP8Review: ` Preffix.

Best regards,
@iszmais

### CODE REVIEW

File: Services/Database/classes/PDO/class.ilDBPdoPostgreSQL.php [hunk 1]
```diff
@@ -25,6 +25,7 @@ public function generateDSN() : void
                      . $this->getUsername() . ';password=' . $this->getPassword();
     }
 
+
```
Comment (mjansenDatabay) [inline Services/Database/classes/PDO/class.ilDBPdoPostgreSQL.php:28]: @iszmais Did the `PHP-CS-FIXER` remove these empty lines? If not please, revert the change (here and everywhere else).l

File: Services/Database/README.md [hunk 2]
```diff
@@ -23,16 +23,16 @@ $n = $DIC->database()->numRows($result);
...
-| float                        | notnull, default                                    | double                                    | number                                                  | 
-| date                         | notnull, default                                    | date                                      | date                                                    | 
-| time                         | notnull, default                                    | time                                      | date                                                    | 
-| timestamp                    | notnull, default                                    | datetime                                  | date                                                    | 
-| clob                         | notnull, default                                    | longtext                                  | clob                                                    | 
+| ilDBInterface Type Supported by ILIAS | Supported Attributes                                | MySQL Mapping                             | Oracle Mapping                                          |
```
Comment (mjansenDatabay) [inline Services/Database/README.md]: @iszmais Please rever this change

**Comment by iszmais** [comment]
@Hi @chfsx ,
This should do as a review. I fixed the most thing mysefl but there are still some part missing (approx. 20 oc). Feel free to ask or write me if you see any misconceptions from my side.

Also, since this is my first review, feel free to leave me a feedback for improvement.

Greetings,
@iszmais

**Comment by mjansenDatabay** [comment]
Maybe we should merge https://github.com/ILIAS-eLearning/ILIAS/pull/3946 before we merge this PR?

**Comment by chfsx** [comment]
I just merged #3946 and made a PR to resolve new conflicts: https://github.com/iszmais/ILIAS/pull/6

**Comment by chfsx** [comment]
Hi @iszmais I merged your and my changes locally, thanks very much!
